### DISCUSSION TITLE
Commandline usage

### DISCUSSION THREAD

**Discussion by avidseeker** [PrivateBin/PrivateBin#1433] [Q&A]
### Describe the problem/question

Is it possible to upload from commandline?

e.g: 

```
$ echo This content is uploaded as a paste | curl -F 'file=@-' https://0x0.st
```

### Did you use the FAQ section?

- [X] Yes, I have read [the FAQ](https://github.com/PrivateBin/PrivateBin/wiki/FAQ) and I found no solution/answer there.

### What you did?

.

### Issue reproducibility

No, I cannot reproduce it on https://privatebin.net.

**Comment by elrido** [comment]
Because the encryption/compression needs to occur on the client side, you would need an extra step in your pipe to handle that. Several CLI clients were written that handle that and the submission (your curl step):
https://github.com/PrivateBin/PrivateBin/wiki/Third-party-clients

**Reply by rugk** [reply]
See also our FAQ: https://github.com/PrivateBin/PrivateBin/wiki/FAQ#can-i-uploaddownload-the-pastes-with-curlwget
